component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/duo/DuoAuthenticator.java

```diff
@@ -279,40 +259,100 @@ private String getErrorPage(AuthenticationContext context) {
     }
 
     /**
-     * Get the mobile claim value of user
+     * Get the mobile claim value of user based on application.authentication.xml configuration.
      *
-     * @param username the username
+     * @param authenticationContext the Authentication Context
      * @return the mobile claim value
      * @throws AuthenticationFailedException
      */
-    private String getMobileClaimValue(String username) throws AuthenticationFailedException {
+    private String getMobileClaimValue(AuthenticationContext authenticationContext)
+            throws AuthenticationFailedException {
+
         String mobileNumber;
-        try {
-            int tenantId = IdentityTenantUtil.getTenantIdOfUser(username);
-            UserRealm userRealm = DuoAuthenticatorServiceComponent.getRealmService().getTenantUserRealm(tenantId);
-            String tenantAwareUsername = MultitenantUtils.getTenantAwareUsername(username);
-            if (userRealm != null) {
-                UserStoreManager userStoreManager = (UserStoreManager) userRealm.getUserStoreManager();
-                mobileNumber = userStoreManager.getUserClaimValue(tenantAwareUsername,
-                        DuoAuthenticatorConstants.MOBILE_CLAIM, null);
-            } else {
-                throw new AuthenticationFailedException(
-                        "Cannot find the user realm for the given tenant: " + tenantId);
+        String username;
+        String federatedMobileNumberAttributeKey;
+        AuthenticatedUser authenticatedUser;
+        Map<ClaimMapping, String> userAttributes;
+        Map<String, String> duoParameters;
+
+        authenticatedUser = (AuthenticatedUser) authenticationContext.getProperty(DuoAuthenticatorConstants.AUTHENTICATED_USER);
+
+        if (authenticatedUser == null) {
+            if (log.isDebugEnabled()) {
+                log.debug("Authentication failed: Could not find the authenticated user. ");
             }
-        } catch (UserStoreException e) {
-            throw new AuthenticationFailedException(DuoAuthenticatorConstants.DuoErrors.ERROR_USER_STORE, e);
+            throw new AuthenticationFailedException
+                    ("Authentication failed: Cannot proceed further without identifying the user. ");
         }
+        username = authenticatedUser.getUserName();
+        duoParameters = FederatedAuthenticatorUtil.getAuthenticatorConfig(DuoAuthenticatorConstants.AUTHENTICATOR_NAME);
+        if (duoParameters != null && duoParameters.get(DuoAuthenticatorConstants.SEND_DUO_TO_FEDERATED_MOBILE_ATTRIBUTE) != null
+                && Boolean.parseBoolean(duoParameters.get(DuoAuthenticatorConstants.SEND_DUO_TO_FEDERATED_MOBILE_ATTRIBUTE))) {
```

Comment:
check the 120 characters limit

Reply:
Fixed in https://github.com/wso2-extensions/identity-outbound-auth-duo/pull/22/commits/655de35d15fe29511f46665ac9dd5e6bf36be362